src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -299,6 +307,43 @@ static void AppendHashtableContents(IndentedTextWriter writer, Hashtable ht)
             }
         }
 
+        private static void EmitScan(IndentedTextWriter writer, RegexMethod rm, string id)
+        {
+            using (EmitBlock(writer, "while (true)"))
+            {
+                using (EmitBlock(writer, "if (FindFirstChar(text))"))
+                {
+                    if (rm.MatchTimeout != int.MaxValue)
+                    {
+                        writer.WriteLine("base.CheckTimeout();");
+                        writer.WriteLine();
+                    }
+
+                    writer.WriteLine("// If we got a match, we're done.");
+                    using (EmitBlock(writer, "if (Go(text))"))
+                    {
+                        writer.WriteLine("return;");
+                    }
+                    writer.WriteLine();
+
+                    writer.WriteLine("// Reset state for another iteration.");
+                    writer.WriteLine("base.runtrackpos = base.runtrack!.Length;");
+                    writer.WriteLine("base.runstackpos = base.runstack!.Length;");
+                    writer.WriteLine("base.runcrawlpos = base.runcrawl!.Length;");
```

Comment:
Are any of these actually necessary?  runcrawlpos should be 0 because if there were captures we would have uncaptured all the way back to 0 on a failed match.  We don't use runtrack at all.   I'd have expected runtrackpos to also end up at runtrack.Length as well, as we use it as a backtracking stack which should get fully unwound.